### PR TITLE
ci: fixes and updates to firebase, build and release actions

### PR DESCRIPTION
**Pull request by moblim**

### CODE REVIEW

**Comment by mirestrepo** [comment]
oh, the firebase task is no longer working... ugh.. I'll check and see

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -36,7 +32,7 @@ jobs:
       - name: Use Node.js
         uses: actions/setup-node@v1
         with:
-          node-version: 12.x
+          node-version: 14.x
```
Comment (RobertGemmaJr) [inline .github/workflows/build.yml:35]: Is there anything specifically preventing us from using 16?

File: .github/workflows/package.yml [hunk 2]
```diff
@@ -22,8 +22,8 @@ jobs:
       fail-fast: false
       matrix:
         os: ${{
-          (github.event.inputs.os == 'all' && fromJSON('["ubuntu-latest", "macOS-latest", "windows-latest"]')) ||
-           (github.event.inputs.os == 'win' && fromJSON('["windows-latest"]')) ||
+          (github.event.inputs.os == 'all' && fromJSON('["ubuntu-latest", "macOS-latest", "windows-2019"]')) ||
+           (github.event.inputs.os == 'win' && fromJSON('["windows-2019"]')) ||
```
Comment (RobertGemmaJr) [inline .github/workflows/package.yml:26]: Hmm... I'm curios what's special about windows-2019? I'm not even sure how Windows managers their versions by year

**Review by RobertGemmaJr — DISMISSED** [review]
I got an "Event Marker not set" warning when I run the executable but afterwards it worked on Windows! Looks great!

File: .github/workflows/package.yml (hunk 2 above)
Comment (mirestrepo) [reply .github/workflows/package.yml:26]: Not fully sure, we just know that this stopped working on new windows runners. I'm not sure if this is still a problem for the newer honeycomb

File: .github/workflows/build.yml (hunk 1 above)
Comment (mirestrepo) [reply .github/workflows/build.yml:35]: I think this task doesn't build in node 16. We could later try upgrading some dependencies and try harder

**Pull request merged**